### PR TITLE
Addon-Storyshots: Adding web-component support

### PR DESCRIPTION
**Pull request by markhuot**
Issue: currently Storyshots can not be used with web-components because Storyshots does not ship with a web-components loader

## What I did

Copied over the HTML loader and changed the appropriate bits to make it load for web-components.

## How to test

All the current Storyshot stories and tests seem to be testing Storyshot functionality against React. There are no existing tests for HTML so I did not add any explicit tests for web-components. Existing tests all pass, but don't really test this new functionality.

- Is this testable with Jest or Chromatic screenshots? n/a
- Does this need a new example in the kitchen sink apps? n/a
- Does this need an update to the documentation? n/a

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shilman** [comment]
Thanks @markhuot, this is awesome!

Any chance you can add a snapshot test to web-components-kitchen-sink to test this in the monorepo? Here's the angular equivalent (and I think the other frameworks also have something similar):

https://github.com/storybookjs/storybook/blob/next/examples/angular-cli/angularshots.test.js

**Comment by markhuot** [comment]
Absolutely! `web-components-storyshots.test.js` is now included in the kitchen sink example. There was some extra noise in the Angluar example that I didn't see in the `vue-kitchen-sink` example so I parsed it back a little. Let me know if you need anything more here!

**Comment by markhuot** [comment]
Doing some more testing here and I'm trying to figure out the best thing for `web-components/renderTree` to return. Should it be a DOM node that Jest can serialize or should it be a raw HTML string? I had a lot of trouble with anything other than a raw HTML string so it currently returns `lit-html.getHTML()` which would be the rendered string of the component/story.

File: addons/storyshots/storyshots-core/src/frameworks/web-components/loader.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  const storybook = jest.requireActual('@storybook/web-components');
+
+  configure({ ...options, storybook });
+
+  return {
+    framework: 'web-components' as const,
+    renderTree: jest.requireActual('./renderTree').default,
+    renderShallowTree: () => {
+      throw new Error('Shallow renderer is not supported for HTML');
```
Comment (igor-dv) [inline addons/storyshots/storyshots-core/src/frameworks/web-components/loader.ts:21]:
```suggestion
      throw new Error('Shallow renderer is not supported for web-components');
```

**Review by igor-dv — COMMENTED** [review]
the reason why storyshots were not supporting web components was that the jsdom was not supporting web components. 
they had a long running issue regarding this https://github.com/jsdom/jsdom/issues/1030.
as far as I remember only jest 26 supports jsdom 16 that should work with web components, though there is a jest-environment-jsdom-sixteen switch for jest 25. I would write this all to docs =)

**Review by ndelangen — APPROVED** [review]

**Comment by markhuot** [comment]
@igor-dv, I added notes to the docs to cover this and the `jsdom` requirements. Let me know how this reads and whether this answers your concern.

**Pull request merged**